### DISCUSSION TITLE
Programmatic way to query notification failure and redeliver message

### DISCUSSION THREAD

**Discussion by liang2zhu1** [community/community#24862] [Apps, API and Webhooks]
<p>For integrator needs a reliable notification on changes on GitHub - is there a programatic way to get the delivery failures and order the re-delivery of the notification? On repo admin UI I can see the past delivery and failure and manually redeliver but I’m not seeing any documentation mentioning a programmatic way.</p>

**Comment by lee-dohm** [comment, marked as answer]
<p>Thanks for reaching out.</p>
<p>Our developer documentation has <a href="https://developer.github.com/v3/guides/best-practices-for-integrators/#favor-asynchronous-work-over-synchronous" rel="nofollow noopener">recommendations for integrators to increase the reliability</a> that you will successfully receive and handle any payload that is sent. We do not provide a mechanism for programmatically resending a payload.</p>
<p>I hope that helps!</p>

**Comment by liang2zhu1** [comment]
<p>&lt;sorry posted the answer twice, remove the first one&gt;</p>

**Comment by liang2zhu1** [comment]
<p>Thank you for the answer - Definitely agree as integrator should do best to increase the reliability but in reality no service can be guaranteed at 100% availability so we sometime have to make reliable system over unreliable infrastructure (like TCP over IP). So my question is, if my requirement is to have a RELIABLE sync from GitHub to my system, it looks I should NOT COMPLETELY rely on GitHub event notification, but instead do a ‘push+pull’ that rely on GitHub notification as the major source for change but a regular query to ensure nothings is missed in the worst case. Does that make sense?</p>

**Comment by lee-dohm** [comment]
<p>Yes, that definitely makes sense.</p>

**Comment by webknjaz** [comment]
<p>Blocker: <a href="https://github.community/t/fr-no-way-to-match-delivered-events-with-ones-queried-via-api/13707">https://github.community/t5/GitHub-API-Development-and/FR-No-way-to-match-delivered-events-with-ones-queried-via-API/m-p/17492#M645</a></p>
